### ISSUE TITLE
Does community support read only?

### ISSUE THREAD

**Issue by 1000copy** [markabrahams/node-net-snmp#100]
I find `agent.addCommunity` function cannot set readonly or readwrite flag .
If I want some community can read write and the other community can read only.What can I do ?

Support I write a agent :

```
var key = "Answer to the Ultimate Question of Life, the Universe, and Everything"
var snmp = require ("../");
var agent = snmp.createAgent({}, function (error, data) {
    if ( error ) {
        console.error (error);
    } else {
        console.log (JSON.stringify(data.pdu.varbinds, null, 2));
    }
});
var authorizer = agent.getAuthorizer ();
authorizer.addCommunity ("public");
var scalarProvider = {
    name: key,
    type: snmp.MibProviderType.Scalar,
    // oid: "1.3.6.1.4.1.66666.1",
    oid: "1",
    scalarType: snmp.ObjectType.OctetString
};
agent.registerProvider (scalarProvider);
var mib = agent.getMib ();
mib.setScalarValue (key, "42");
mib.dump ()
```

I expect and I can :

`# snmpget -v2c -cpublic 127.0.0.1 .1.0`

I can write it too.But actually ,I just want it be readonly:

`# snmpset -v2c -cpublic 127.0.0.1 .1.0 s "43"`

**Comment by markabrahams** [comment]
Hi @1000copy - there is no read-only community/USM support at the moment.  This would be a useful enhancement though.

**Comment by markabrahams** [comment]
Hi @1000copy - I've added read-only access support to the agent and published this in version 2.9.0 of the npm.

README.md has details, but here's some example usage:

```
    var agent = snmp.createAgent({
        accessControlModelType: snmp.AccessControlModelType.Simple
    }, function (error, data) {
        // null callback for example brevity
    });
    var authorizer = agent.getAuthorizer ();
    authorizer.addCommunity ("public");
    authorizer.addCommunity ("private");
    authorizer.addUser ({
        name: "fred",
        level: snmp.SecurityLevel.noAuthNoPriv
    });
    var acm = authorizer.getAccessControlModel ();
    // Since read-only is the default, explicitly setting read-only access is not required - just shown here as an example
    acm.setCommunityAccess ("public", snmp.AccessLevel.ReadOnly);
    acm.setCommunityAccess ("private", snmp.AccessLevel.ReadWrite);
    acm.setUserAccess ("fred", snmp.AccessLevel.ReadWrite);

```
